webapi/Skills/ChatSkills/ChatSkill.cs

```diff
@@ -215,47 +216,82 @@ public async Task<string> ExtractAudienceAsync(SKContext context, CancellationTo
     /// <summary>
     /// Extract chat history.
     /// </summary>
-    /// <param name="context">Contains the 'tokenLimit' controlling the length of the prompt.</param>
     /// <param name="cancellationToken">The cancellation token.</param>
     [SKFunction, Description("Extract chat history")]
-    public async Task<string> ExtractChatHistoryAsync(
+    public async Task<string> ExtractChatHistoryAsStringAsync(
         [Description("Chat ID to extract history from")] string chatId,
         [Description("Maximum number of tokens")] int tokenLimit,
         CancellationToken cancellationToken = default)
+    {
+        return await this.ExtractChatHistoryAsync(chatId, tokenLimit, cancellationToken: cancellationToken);
+    }
```

Comment:
Feel free to use this or not but I just wanted to let you know you can also just do the following:

```suggestion
    public Task<string> ExtractChatHistoryAsStringAsync(string chatId, int tokenLimit,
        CancellationToken cancellationToken = default)
    {
        return this.ExtractChatHistoryAsync(chatId, tokenLimit, cancellationToken: cancellationToken);
    }
```

This just makes the caller await instead of us awaiting and then the caller await on us. It's pretty much the same except for where exceptions would be thrown.

Reply:
I see! This method is called from a semantic function (so in a rendered prompt). I'm not sure what behavior is there, if the caller awaits async functions or not. Do you know? And would it change your suggestion?

Reply:
It does not make a difference for the callers.  Either way, the caller will have to await.

The only difference is that in one case, they'll await on the inner call's Task passed straight through ExtractChatHistoryAsStringAsync() and in the other case, ExtractChatHistoryAsStringAsync() is the one that awaits the inner call while the called awaits ExtractChatHistoryAsStringAsync().

It's not much of a difference either way.  I just wanted you to be aware that either can be done.

It's mostly a matter for exception handling.

For more info:
https://code-maze.com/charp-difference-between-returning-and-awaiting-a-task/

Reply:
Thanks for explaining! I changed it per your request since we mostly let exceptions bubble up anyway.